### PR TITLE
Fixes sprint 5

### PR DESCRIPTION
**Pull request by ignaciocervinoGlob**
### Description
Fixing bugs in the application.

### Tasks
- [x] Task 1 → When the add form comes from scanner, it must disable barcode field and hide confirmation button
- [x] Task 2 → In confirmation dialog, when there's no internet show warning (not success because it's not changing DB) 
- [x] Task 3 → Check why takes too long to return to main after successfully save
- [x] Task 4 → Barcode check button in addForm checks for containing string and not unique one

### Evidences → iPhone 11
| Task 1 | Task 2 | Task 3 |
|---------------|-------------|-------------|
| image | Simulator Screen Recording - iPhone 11 - 2022-03-02 at 12 07 27 | Simulator Screen Recording - iPhone 11 - 2022-03-02 at 12 16 54 |

### CODE REVIEW

**Comment by ezequielvelezglobant** [comment]
@ignaciocervinoGlob review behaviour.
Task 1: you hide confirmation button but message said that I have to press the button to check if exists.
Task 2: if the user lost internet the dialog appear for ever? I guess that is better to have a "retry" and "cancel" cta.

**Comment by ignaciocervinoGlob** [comment]
> @ignaciocervinoGlob review behaviour. Task 1: you hide confirmation button but message said that I have to press the button to check if exists. Task 2: if the user lost internet the dialog appear for ever? I guess that is better to have a "retry" and "cancel" cta.

Task 1:  Fixed.

Task 2: if user lost internet in confirmation dialog, this one dismiss and user has to reconnect to internet to save the asset successfully.
It's possible to do that modification, but we are following Android behavior that was already approved.

**Review by ezequielvelezglobant — APPROVED** [review]

File: stockApp/stockApp/Presentation/View/FormViewController.swift [hunk 1]
```diff
@@ -147,8 +150,11 @@ class FormViewController: UIViewController, UIScrollViewDelegate {
 
     private func validationSuccess() -> (_ alertAction: UIAlertAction) -> Void {
         return { [weak self] _ in
-            self?.viewModel.saveAsset()
-            self?.dismiss(animated: false, completion: nil)
+            if Environment.current.reachabilityMonitor.isNotConnectionInternet {
```
Comment (fn1983) [inline stockApp/stockApp/Presentation/View/FormViewController.swift:153]: is this something related to View ? 🤔

Comment (ignaciocervinoGlob) [reply stockApp/stockApp/Presentation/View/FormViewController.swift:153]: We call a function in a singleton that tells us if there is internet, maybe this validation can be done in the VM. I will take this suggestion for the future
